datalake_api/fetcher.py

```diff
@@ -37,20 +40,22 @@ def _adjusted_path(self):
             return m.group('path')
         return self.metadata['path']
 
-    def _guess_type(self):
-        self.content_type, self.content_encoding = \
-            guess_type(self._adjusted_path)
-        if self.content_type is None and self._is_log_file():
-            self.content_type = 'text/plain'
+    @memoized_property
+    def content_type(self):
+        _type, _encoding = guess_type(self._adjusted_path)
+        return _type
+
+    @memoized_property
+    def content_encoding(self):
+        mime = magic.from_buffer(self._header, mime=True)
+        if mime == 'application/gzip':
+            return 'gzip'
+        return None
 
     _KNOWN_LOGS = ['syslog', 'dmesg']
 
-    def _is_log_file(self):
-        return self.metadata['what'] in self._KNOWN_LOGS or \
-            self._adjusted_path.endswith('.log') or \
-            self._adjusted_path.endswith('.log.gz') or \
-            self._adjusted_path.endswith('.log.1.gz') or \
-            self._adjusted_path.endswith('.log.1')
+    def read(self):
```

Comment:
Ah ok yeah, that makes total sense. I don't think it's blocking this diff, because as you say it's internal and still a huge improvement over matching filenames. Perhaps just rename it to read_stream() or something so it doesn't get confused with a file-like object?